### ISSUE TITLE
ipc port is no longer valid (early)

### ISSUE THREAD

**Issue by DangerD1024** [fcpxhacks/fcpxhacks#3375]
Script stops sometimes on line `log.ef("W0")` :
(function is being called from Terminal)
```
	local success, errorMessage = pcall(function()
	...
		log.ef("111")
		isBusy = true;
		while isBusy do
			log.ef("W0")
			local backgroundTasksDialog = fcp.backgroundTasksDialog
```
```
2024-09-23 15:45:44: 15:45:44 ERROR:   example: 111
2024-09-23 15:45:44: 15:45:44 ERROR:   example: Oh Crap ! Error: ...CommandPost.app/Contents/Resources/extensions/hs/ipc.lua:48: ipc port is no longer valid (early)
2024-09-23 15:45:44: 15:45:44 ERROR:   LuaSkin: hs.ipc:callback - error during callback for CommandPost: ...CommandPost.app/Contents/Resources/extensions/hs/ipc.lua:420: ipc port is no longer valid (early)
stack traceback:
	[C]: in method 'sendMessage'
	...CommandPost.app/Contents/Resources/extensions/hs/ipc.lua:420: in function 'hs.ipc.__defaultHandler'
```

Any ideas?

**Comment by latenitefilms** [comment]
Sorry for the delayed replay - missed this.

Normally you can ignore the `ipc port is no longer valid (early)` message.

But in this case, I'm guessing your CLI call is closing once the Lua code is done?

**Comment by latenitefilms** [comment]
@DangerD1024 - Are you still having issues?

**Comment by latenitefilms** [comment]
I'm going to close this issue for now, but feel free to re-open it if you're still having issues with the latest CommandPost release.

**Comment by DangerD1024** [comment]
> I'm going to close this issue for now, but feel free to re-open it if you're still having issues with the latest CommandPost release.

Yes, i still have issues, is there any way to increase timeout for scripts started from command line tool? Currently it just stops execution...

**Comment by DangerD1024** [comment]
I still have issue, if i call function from commandpost console, it works fine, but if i call it from terminal:
`cmdpost -c 'startExport("/Users/realmedia/Documents/FCP Projects/257 York Road, Dundas, testOLOLO_unbranded.fcpxml")'`
It stops in the middle of script randomly even between print messages...

i'm still on version 1.4.27 because my script which i've just finished stopped to work on 1.5...

**Comment by DangerD1024** [comment]
UPD on latest 1.5.1 script works, looks like something was fixed, but same issue from terminal remains

**Comment by DangerD1024** [comment]
Any ideas?
Alternatively i can read some txt file each 5 seconds and get tasks from there =)

**Comment by DangerD1024** [comment]
Yes, that works perfect =)
